src/components/GlobalLeaderboards/Context.tsx

```diff
@@ -0,0 +1,213 @@
+  period: PerfPeriod.DAY,
+  setPeriod: () => {},
+  setAccountsOrderBy: () => {},
+  setAccountsOrderDirection: () => {},
+  setPositionsOrderBy: () => {},
+  setPositionsOrderDirection: () => {},
+  topAccountsHeaderClick: () => () => {},
+  topPositionsHeaderClick: () => () => {},
+});
+
+type ProfilerRef = ReturnType<typeof Profiler> & {
+  positions?: boolean;
+  accounts?: boolean;
+  pSorted?: boolean;
+  aSorted?: boolean;
+  pReturned?: boolean;
+  aReturned?: boolean;
+  reported?: boolean;
+}
+
+export const useLeaderboardContext = () => useContext(LeaderboardContext);
+
+export const LeaderboardContextProvider: FC<PropsWithChildren> = ({ children }) => {
+  const p = useRef<ProfilerRef>();
+  if (!p.current) {
+    p.current = Profiler();
+  }
+  const { chainId } = useChainId();
+  const [period, setPeriod] = useState<PerfPeriod>(PerfPeriod.TOTAL);
+  const [accountsOrderBy, setAccountsOrderBy] = useState<keyof TopAccountsRow>("absPnl");
+  const [accountsOrderDirection, setAccountsOrderDirection] = useState<number>(1);
+  const [positionsOrderBy, setPositionsOrderBy] = useState<keyof TopPositionsRow>("unrealizedPnl");
+  const [positionsOrderDirection, setPositionsOrderDirection] = useState<number>(1);
+  const topPositionsUnordered = useTopPositions();
+  if (
+    topPositionsUnordered &&
+    topPositionsUnordered.data &&
+    topPositionsUnordered.data.length &&
+    p.current && !p.current.positions
+  ) {
+    p.current("topPositionsUnordered");
+    p.current.positions = true;
+  }
+  const topAccountsUnordered = useTopAccounts(period);
+  if (
+    topAccountsUnordered &&
+    topAccountsUnordered.data &&
+    topAccountsUnordered.data.length &&
+    p.current && !p.current.accounts
+  ) {
+    p.current("topAccountsUnordered");
+    p.current.accounts = true;
+  }
+  const [topPositions, setTopPositions] = useState<Array<TopPositionsRow>>([]);
+  const [topAccounts, setTopAccounts] = useState<Array<TopAccountsRow>>([]);
+  const [positionsHash, setPositionsHash] = useState<string>();
+  const [accountsHash, setAccountHash] = useState<string>();
+  const topAccountsHeaderClick = useCallback((key: keyof TopAccountsRow) => () => {
+    if (key === "wins") {
+      setAccountsOrderBy(accountsOrderBy === "wins" ? "losses" : "wins");
+      setAccountsOrderDirection(1);
+    } else if (key === accountsOrderBy) {
+      setAccountsOrderDirection((d: number) => -1 * d);
+    } else {
+      setAccountsOrderBy(key);
+      setAccountsOrderDirection(1);
+    }
+  }, [accountsOrderBy]);
+
+  const topPositionsHeaderClick = useCallback((key: keyof TopPositionsRow) => () => {
+    if (key === positionsOrderBy) {
+      setPositionsOrderDirection((d: number) => -1 * d);
+    } else {
+      setPositionsOrderBy(key);
+      setPositionsOrderDirection(1);
+    }
+  }, [positionsOrderBy]);
+
+  const nextPositionsHash = utils.sha256(topPositionsUnordered.data ? topPositionsUnordered.data.map(a => (
```

Comment:
1. Could you please explain the purpose of using hash there?
2. What the perfomance of the hash function, probably we should use it inside useMemo?